src/System.Text.Json/tests/JsonObjectTests.cs

```diff
@@ -712,11 +654,147 @@ public static void TestArgumentNullValidation()
         }
 
         [Fact]
-        public static void TestDuplicatesEnumOutOfRange()
+        public static void TestStringComparisonEnum()
+        {
+            var jsonObject = new JsonObject()
+            {
+                { "not encyclopaedia", "value1" },
+                { "Encyclopaedia", "value2" },
+                { "NOT encyclopaedia", "value3" },
+                { "encyclopaedia", "value4" }
+            };
+
+            Assert.Equal(4, jsonObject.Count());
+
+            Assert.False(jsonObject.ContainsProperty("ENCYCLOPAEDIA"));
+            Assert.False(jsonObject.TryGetPropertyValue("ENCYCLOPAEDIA", out JsonNode jsonNode));
+            Assert.Null(jsonNode);
```

Comment:
**Question:** Do we have a test where the property we are searching for has the value "null" so TryGet returns true and JsonNode is not null, but rather is a JsonNull? If so, great. Otherwise, let's add it.

Reply:
We have such test for indexer, I can add it for Get/(Try)Get as well.